app/models/host/discovered.rb

```diff
@@ -86,9 +95,27 @@ def populate_fields_from_facts facts = self.facts_hash, type = 'puppet'
       importer = super(facts)
     end
     self.subnet = Subnet.subnet_for(importer.ip)
+    self.discovery_attribute_set = DiscoveryAttributeSet.where(:host_id => id).first_or_create
+    self.discovery_attribute_set.update_attributes(import_from_facts)
     self.save
   end
 
+  def import_from_facts(facts = self.facts_hash)
+    cpu_count  = facts['physicalprocessorcount'].to_i
```

Comment:
One last nitpick - can you add a rescue branch just in case someone sends wrong input?

    cpu_count  = facts['physicalprocessorcount'].to_i rescue 0

The same for memory and perhaps in the each block for the disks size. Converting to numbers can fail hard.